### PR TITLE
Added a new dropdown at navbar and details page

### PR DESCRIPTION
**Pull request by imhardikdesai**
I have added dropdown in navbar for features/tools #357 
 
image

And also added details page for this

### CODE REVIEW

**Comment by mohitkhatri06** [comment]
@imhardikdesai when I scroll in any feature and then try to go to other feature it starts from the bottom

File: src/components/UnAuthenticatedHeader/UnAuthenticatedHeader.tsx [hunk 1]
```diff
@@ -11,15 +11,43 @@ import {
...
 import { useAuthContext } from '@/hooks/use-auth'
-import { CALCULATOR_ROUTES } from '@/utils/app-shell'
+import { CALCULATOR_ROUTES, FEATURES_ROUTES } from '@/utils/app-shell'
 import { cn } from '@/utils/utils'
 import When from '../When/When'
 import { buttonVariants } from '../ui/button'
 
-const ListItem = React.forwardRef<
+const ListItemFeatures = React.forwardRef<
   React.ElementRef<'a'>,
```
Comment (rockingrohit9639) [inline src/components/UnAuthenticatedHeader/UnAuthenticatedHeader.tsx:20]: why do we need `forwardRef` here ?

**Review by rockingrohit9639 — CHANGES_REQUESTED** [review]

**Comment by imhardikdesai** [comment]
@mohitkhatri06 @rockingrohit9639 i will fix the changes

**Comment by imhardikdesai** [comment]
Hello @mohitkhatri06 and @rockingrohit9639 Please review above changes and let me now if anything else

File: src/pages/Features/FeaturesDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      break
+    case 'balance-sheet':
+      toolDetailComponent = <BalanceSheetDetails />
+      break
+    case 'calculators':
+      toolDetailComponent = <Calculators />
+      break
+    default:
+      toolDetailComponent = <NotFound />
+      break
```
[the remark this answers is not in the document]
Comment (mohitkhatri06) [reply src/pages/Features/FeaturesDetails.tsx:34]: @imhardikdesai can you have a look on this file?

[the remark this answers is not in the document]
Comment (imhardikdesai) [reply src/pages/Features/FeaturesDetails.tsx:34]: Yes @mohitkhatri06 I have wrapped logic inside useMemo hook as Rohit says

[the remark this answers is not in the document]
Comment (imhardikdesai) [reply src/pages/Features/FeaturesDetails.tsx:34]: Do you think I have to define route for each components in App.js??

[the remark this answers is not in the document]
Comment (mohitkhatri06) [reply src/pages/Features/FeaturesDetails.tsx:34]: @rockingrohit9639 what do you think, i guess we can proceed now?

**Review by mohitkhatri06 — APPROVED** [review]

**Review by rockingrohit9639 — DISMISSED** [review]
He has not updated what I have said but @mohitkhatri06 wanted me to approve it so I here I am.

**Review by rockingrohit9639 — APPROVED** [review]

**Pull request merged**